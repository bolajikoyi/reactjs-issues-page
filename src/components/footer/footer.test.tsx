import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Footer from './footer';

afterEach(cleanup)

describe('Footer Component', ()=>{
    test('expect component to be created', () => {
        expect(Footer).toBeTruthy()
    });
    
    test('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Footer />, div);
    });
    
    test("Should Contain Text 'Blog'", async () => {
        render(<Footer />);
        screen.getByText('Blog');
    });

    test("Should Contain span tag for Blog text", async () => {
        render(<Footer />);
        expect(screen.getByText("Blog").tagName).toBe("SPAN");
    });

    test("Should Contain ClassName", async () => {
        render(<Footer />);
        expect(screen.getByText("Blog").className).toBe(
          "blog"
        );  
      });

    test('matches snapshot', ()=> {
        const tree = renderer.create(<Footer/>).toJSON();
        expect(tree).toMatchSnapshot();
    })

})

