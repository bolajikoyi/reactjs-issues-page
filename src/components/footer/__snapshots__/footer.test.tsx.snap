// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component matches snapshot 1`] = `
<div>
  <div
    className="vertical-line"
  />
  <div
    className="Footer-container"
  >
    <span>
      <i
        className="footer fab fa-github"
      />
      <span
        className="Footer-copyright"
      >
        Â© 
        2022
         Github, Inc.
      </span>
    </span>
    <div
      className="Footer-links"
    >
      <span>
        Terms
      </span>
      <span>
        Privacy
      </span>
      <span>
        Security
      </span>
      <span>
        Status
      </span>
      <span>
        Docs
      </span>
      <span>
        Contact GitHub
      </span>
      <span>
        Pricing
      </span>
      <span>
        API
      </span>
      <span>
        Training
      </span>
      <span
        className="blog"
      >
        Blog
      </span>
      <span>
        About
      </span>
    </div>
  </div>
</div>
`;
